supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -401,49 +488,97 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
+   * looks for a previous state on the handle; if it doesn't find anything (task just started), we poll for the status
+   * and also fetch the cost data. If there is a previous status, we look and see if the cost data has been fetched.
+   * If not, we poll for the status AND cost information in TES.
+   * */
+  override def pollStatusAsync(handle: StandardAsyncPendingExecutionHandle): Future[TesRunStatus] = {
+    val fetchCostData = !handle.previousState.flatMap(_.costData).map(_.fullyPopulated).getOrElse(false)
+    pollTesStatus(handle,
+                  fetchCostData,
+                  fetchFullTaskView,
+                  fetchMinimalTesTask,
+                  getTesStatus,
+                  tellMetadata,
+                  getErrorLogs
+    )
 
-          case s if s.contains("EXECUTOR_ERROR") =>
-            jobLogger.info(s"TES reported a failure for Job ${handle.pendingJob.jobId}: '$s'")
-            Failed()
+  }
 
-          case s if s.contains("SYSTEM_ERROR") =>
-            jobLogger.info(s"TES reported an error for Job ${handle.pendingJob.jobId}: '$s'")
-            Error()
+  private def getTesStatus(state: Option[String], withCostData: Option[TesVmCostData], jobId: String): TesRunStatus =
+    state match {
+      case s if s.contains("COMPLETE") =>
+        jobLogger.info(s"Job ${jobId} is complete")
+        Complete(withCostData)
 
-          case _ => Running
-        }
+      case s if s.contains("CANCELED") =>
+        jobLogger.info(s"Job ${jobId} was canceled")
+        Cancelled(withCostData)
+
+      case s if s.contains("EXECUTOR_ERROR") =>
+        jobLogger.info(s"TES reported a failure for Job ${jobId}: '$s'")
+        Failed()
+
+      case s if s.contains("SYSTEM_ERROR") =>
+        jobLogger.info(s"TES reported an error for Job ${jobId}: '$s'")
+        Error()
+
+      case _ => Running(withCostData)
     }
 
-  private def getErrorLogs(handle: StandardAsyncPendingExecutionHandle): Future[Seq[String]] =
-    makeRequest[Task](HttpRequest(uri = s"$tesEndpoint/${handle.pendingJob.jobId}?view=FULL")) map { response =>
-      response.logs.flatMap(_.lastOption).flatMap(_.system_logs).getOrElse(Seq.empty[String])
+  private def getErrorLogs(handle: StandardAsyncPendingExecutionHandle): Future[Seq[String]] = {
+    val task = fetchFullTaskView(handle)
+    task.map(t => t.logs.flatMap(_.lastOption).flatMap(_.system_logs).getOrElse(Seq.empty[String]))
+  }
+
+  private def getTaskLogs(handle: StandardAsyncPendingExecutionHandle): Future[TaskLog] = {
+    val task = fetchFullTaskView(handle)
+    val errorStates = List("EXECUTOR_ERROR", "SYSTEM_ERROR")
+
+    val stateAndLogs = for {
+      task <- task
+      state = task.state
+      taskLog = task.logs
+    } yield (state, taskLog)
+
+    stateAndLogs map { case (state, taskLog) =>
+      if (errorStates.contains(state)) {
+        Future.failed(new RuntimeException(s"Failed TES request: $state"))
+      }
+      taskLog.flatMap(_.headOption).head
```

Comment:
I think this `.head` will crash if the `Seq[TaskLog]` is empty. Maybe this function should return a Future[Option[TaskLog]] since it might not always be able to retrieve one?